File: greedy/1518_water_bottles/1518_water_bottles.go
package main

import "fmt"

func numWaterBottles(numBottles int, numExchange int) int {
	s := 0
	left := 0
	for numBottles > 0 {
		s += numBottles
		left += numBottles
		numBottles = left / numExchange
		left %= numExchange
	}
	return s
}

func main() {
	numBottles := 9
	numExchange := 3
	res := numWaterBottles(numBottles, numExchange)
	fmt.Println(res)
}

File: greedy/1518_water_bottles/1518_water_bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = numBottles
        while numBottles >= numExchange:
            cnt = numBottles // numExchange
            res += cnt
            numBottles = numBottles - (cnt * numExchange)
            numBottles += cnt
        return res


numBottles = 9
numExchange = 3
res = Solution().numWaterBottles(numBottles, numExchange)
print(res)